homeassistant/components/glances/sensor.py

```diff
@@ -344,6 +354,24 @@ def available(self) -> bool:
     @property
     def native_value(self) -> StateType:
         """Return the state of the resources."""
+        if self.entity_description.type == "uptime":
+            uptime = self.coordinator.data[self.entity_description.type]
+            if uptime.find("day") > -1:
+                uptime = datetime.strptime(uptime.replace("s", ""), "%d day %H:%M:%S")
+                uptime_delta = timedelta(
+                    days=uptime.day,
+                    hours=uptime.hour,
+                    minutes=uptime.minute,
+                    seconds=uptime.second,
+                )
+            else:
+                uptime = datetime.strptime(uptime, "%H:%M:%S")
+                uptime_delta = timedelta(
+                    hours=uptime.hour, minutes=uptime.minute, seconds=uptime.second
+                )
+
+            self.coordinator.data["uptime"] = {"uptime": dt_util.now() - uptime_delta}
```

Comment:
Do you need to update the data in the source structure ?
I would move this whole block after the generic "value = .." and only update the value, like this :

value = self.coordinator.data[self.entity_description.type]

if self.entity_description.type == "uptime":
    value = convert_uptime_to_timestamp(value)